File: sling-react-core/src/main/java/com/cathcart93/sling/models/GreetingController.java
package com.cathcart93.sling.models;

import com.cathcart93.sling.BaseController;
import com.google.gson.annotations.Expose;
import org.apache.sling.api.resource.Resource;
import org.apache.sling.models.annotations.Model;
import org.apache.sling.models.annotations.injectorspecific.ValueMapValue;

/**
 * Created by Kusak on 7/15/2017.
 */
@Model(adaptables = Resource.class)
public class GreetingController extends BaseController {

    @Expose
    private String __type = "Greeting";

    @Expose
    @ValueMapValue(name = "message", optional = true)
    private String message;

}

File: sling-react-components/src/main/kotlin/com/cathcart93/sling/components/models/GreetingController.kt
package com.cathcart93.sling.components.models

import com.cathcart93.sling.core.IReactController
import com.cathcart93.sling.core.ReactController
import com.cathcart93.sling.core.ReactProp
import org.apache.sling.api.resource.Resource
import org.apache.sling.models.annotations.Model
import org.apache.sling.models.annotations.injectorspecific.ValueMapValue

/**
 * Created by Kusak on 7/15/2017.
 */
@Model(adaptables = arrayOf(Resource::class))
@ReactController(componentName = "Greeting")
class GreetingController : IReactController {

    @ValueMapValue(name = "message", optional = true)
    @ReactProp
    private val message: String? = null

}
